src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -298,6 +306,9 @@ def app_deploy(
         if recursive is None:
             recursive = False
 
+    if has_paths and prune:
```

Comment:
Is all the above a good candidate for expanding the policies we wrote earlier for capturing combinations of flags? Not a holdup for this PR though.

Reply:
I don't think that falls under a policy. It's just mutually exclusive flags that we'll end up allowing in the future I think, but there needs significant work before we can do that.